### ISSUE TITLE
[RFC] Announcement service for aria-live

### ISSUE THREAD

**Issue by Stanton** [jadu/pulsar#1161]
There is a common need for certain interactions—usually javascript driven—to provide information to assistive technologies to communicate specific changes of state.

An example may be an inline form which adds an item to a list without refreshing the page, it is desirable for these interactions to update a live region so that AT will announce a useful message such as '$Item added'. Currently we rely on each component to be responsible for supporting live regions, which is moderately successful for brand new components, but means that we often can not, or do not retrofit this behaviour into existing components.

An idea was bounced about which would lead to us creating a new Pulsar component which would provide easy methods to trigger a new aria-live announcement.

# Example

```
pulsar.announce('This is a live message');
```

We may decide to support live/assertive in a similar way that flash messages use `pulsar.flash.success('this is a flash message')`.

We would need to write this in such a way that any JS code (whether part of the jQuery + browserify bundle, or plain old vanilla legacy scripts) across the platform can trigger an announcement.

Open for early stage discussion, I've not focused on how the live regions should be created/updated as part of this description but any suggestions are welcome in the comments.

**Comment by Stanton** [comment]
First attempt at implementing this https://github.com/jadu/pulsar-live-region
